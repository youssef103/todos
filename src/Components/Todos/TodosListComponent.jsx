import React, { useEffect, useCallback } from "react";
import Status from "../../Common/Status";
import todosStyle from "./todos.module.scss";

export const TodosListComponent = props => {
  // eslint-disable-next-line
  const fetchTodos = useCallback(async () => {
    props.getAllTodos();
  });

  useEffect(() => {
    fetchTodos();

    return () => {
      console.log("cleanup");
    };

    // eslint-disable-next-line
  }, []);

  return (
    <Status {...props}>
      <div className={todosStyle.Todos}>
        {props.todos.length === 0 && <p>Todos list is empty</p>}
        {props.todos.map(todo => (
          <label htmlFor={todo.id} className={todosStyle.Item} key={todo.id}>
            <span>{todo.title}</span>
            <span>
              <input type="checkbox" name={todo.id} id={todo.id} />
            </span>
          </label>
        ))}
      </div>
    </Status>
  );
};
