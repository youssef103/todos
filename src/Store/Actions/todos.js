import {
  FETCH_TODOS,
  FETCH_TODOS_SUCCESS,
  FETCH_TODOS_FAILURE
} from "../Types";
import api from "../../Services/api";


export const getTodos = () => async dispatch => {
  dispatch(fetchTodos());

  try {
    const todos = await api.getAll("todos");
    //const todos = process.env.NODE_ENV !== 'test' ? await api.getAll("todos") : data;
    dispatch(fetchTodosSuccess(todos));
  } catch (error) {
    dispatch(fetchTodosFailure(error));
  }
};

export const fetchTodos = () => {
  return {
    type: FETCH_TODOS
  };
};

export const fetchTodosSuccess = todos => {
  return {
    type: FETCH_TODOS_SUCCESS,
    todos
  };
};

export const fetchTodosFailure = error => {
  return {
    type: FETCH_TODOS_FAILURE,
    notification: {
      type: "error",
      message: error.message
    }
  };
};