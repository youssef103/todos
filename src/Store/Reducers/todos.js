import {
  FETCH_TODOS,
  FETCH_TODOS_SUCCESS,
  FETCH_TODOS_FAILURE
} from "../Types";

const initialState = {
  todos: [],
  loading: false,
  notification: {
    type: "",
    message: ""
  }
};

export const todosReducer = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_TODOS:
      return {
        ...state,
        loading: true
      };
    case FETCH_TODOS_SUCCESS:
      return {
        ...state,
        todos: action.todos,
          loading: false
      };
    case FETCH_TODOS_FAILURE:
      return {
        ...state,
        notification: {
            type: action.notification.type,
            message: action.notification.message
          },
          loading: false
      };
    default:
      return state;
  }
};