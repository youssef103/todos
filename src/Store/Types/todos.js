export const FETCH_TODOS = 'FETCH_TODOS';
export const FETCH_TODOS_SUCCESS = 'FETCH_TODOS_SUCCESS';
export const FETCH_TODOS_FAILURE = 'FETCH_TODOS_FAILURE';