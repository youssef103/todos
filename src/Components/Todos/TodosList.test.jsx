import React from "react";
import { shallow } from "enzyme";
import { TodosListComponent } from "./TodosListComponent";
import { todos } from "../../../fixture/todos";

beforeEach(() => {});

test("should render TodosList correctly", () => {
  const wapper = shallow(<TodosListComponent todos={todos} />);

  expect(wapper).toMatchSnapshot();
});

test("should render TodosList with empty message", () => {
  const wapper = shallow(<TodosListComponent todos={[]} />);

  expect(wapper).toMatchSnapshot();
});
