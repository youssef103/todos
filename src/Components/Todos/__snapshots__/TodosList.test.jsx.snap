// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TodosList correctly 1`] = `
<Status
  todos={
    Array [
      Object {
        "completed": false,
        "id": 1,
        "title": "delectus aut autem",
        "userId": 1,
      },
      Object {
        "completed": false,
        "id": 2,
        "title": "quis ut nam facilis et officia qui",
        "userId": 1,
      },
      Object {
        "completed": false,
        "id": 3,
        "title": "fugiat veniam minus",
        "userId": 1,
      },
      Object {
        "completed": true,
        "id": 4,
        "title": "et porro tempora",
        "userId": 1,
      },
      Object {
        "completed": false,
        "id": 5,
        "title": "laboriosam mollitia et enim quasi adipisci quia provident illum",
        "userId": 1,
      },
      Object {
        "completed": false,
        "id": 6,
        "title": "qui ullam ratione quibusdam voluptatem quia omnis",
        "userId": 1,
      },
    ]
  }
>
  <div
    className="Todos"
  >
    <label
      className="Item"
      htmlFor={1}
      key="1"
    >
      <span>
        delectus aut autem
      </span>
      <span>
        <input
          id={1}
          name={1}
          type="checkbox"
        />
      </span>
    </label>
    <label
      className="Item"
      htmlFor={2}
      key="2"
    >
      <span>
        quis ut nam facilis et officia qui
      </span>
      <span>
        <input
          id={2}
          name={2}
          type="checkbox"
        />
      </span>
    </label>
    <label
      className="Item"
      htmlFor={3}
      key="3"
    >
      <span>
        fugiat veniam minus
      </span>
      <span>
        <input
          id={3}
          name={3}
          type="checkbox"
        />
      </span>
    </label>
    <label
      className="Item"
      htmlFor={4}
      key="4"
    >
      <span>
        et porro tempora
      </span>
      <span>
        <input
          id={4}
          name={4}
          type="checkbox"
        />
      </span>
    </label>
    <label
      className="Item"
      htmlFor={5}
      key="5"
    >
      <span>
        laboriosam mollitia et enim quasi adipisci quia provident illum
      </span>
      <span>
        <input
          id={5}
          name={5}
          type="checkbox"
        />
      </span>
    </label>
    <label
      className="Item"
      htmlFor={6}
      key="6"
    >
      <span>
        qui ullam ratione quibusdam voluptatem quia omnis
      </span>
      <span>
        <input
          id={6}
          name={6}
          type="checkbox"
        />
      </span>
    </label>
  </div>
</Status>
`;

exports[`should render TodosList with empty message 1`] = `
<Status
  todos={Array []}
>
  <div
    className="Todos"
  >
    <p>
      Todos list is empty
    </p>
  </div>
</Status>
`;
